lib/src/main/scala/spinal/lib/com/jtag/JtagTapInstructions.scala

```diff
@@ -0,0 +1,145 @@
+package spinal.lib.com.jtag
+
+import spinal.core._
+import spinal.lib._
+
+// Created by PIC32F_USER on 05/04/2016.
+// Modified / Extended by HWEngineer 15/03/2020
+
+//══════════════════════════════════════════════════════════════════════════════
+// define JtagTapInstruction
+//
+class JtagTapInstruction(tap: JtagTapAccess, val instructionId: Int) extends Area {
+  def doCapture(): Unit = {}
+  def doShift(): Unit = {}
+  def doUpdate(): Unit = {}
+  def doReset(): Unit = {}
+
+  val instructionHit = tap.getInstruction === B(instructionId, tap.getInstruction.getWidth bits)
+
+  Component.current.addPrePopTask(() => {
```

Comment:
I'm thinking about removing that, and having the callback managed from the jtag tap itself, this would avoid having the JtagTapInstruction having to know the the state of the jtag